### PR TITLE
EVG-657 Add naive auth

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: user_service_basic_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	assert.Implements((*UserManager)(nil), &NaiveUserManager{})
+
+	u, err := NewNaiveUserManager([]*NaiveUser{
+		{
+			User:         "foo",
+			Pass:         "bar",
+			EmailAddress: "baz",
+		},
+	})
+	user, err := u.GetUserByToken(context.Background(), "0:baz:[56 88 246 34 48 172 60 145 95 48 12 102 67 18 198 63]")
```
Comment (julianedwards) [inline user_service_basic_test.go]: I would make the token string a constant: `expectedToken := fmt.Sprintf("0:baz:%v", md5.Sum([]byte("foo"+"bar")))`.

Comment (bsamek) [reply user_service_basic_test.go]: Done.

File: user_service_basic.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import (
+	"context"
+	"crypto/md5"
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+type NaiveUser struct {
```
Comment (tychoish) [inline user_service_basic.go]: can you use basic user here?

File: user_service_basic.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+func (u *NaiveUser) GetAPIKey() string   { return "" }
+func (u *NaiveUser) Roles() []string     { return []string{} }
+
+// NaiveUserManager implements the UserManager interface and has a list of
+// NaiveUsers which is stored in the settings configuration file.
+// Note: This use of the UserManager is recommended for dev/test purposes only
+// and users who need high security authentication  mechanisms should rely on a
+// different authentication mechanism.
+type NaiveUserManager struct {
+	users []*NaiveUser
```
Comment (tychoish) [inline user_service_basic.go]: slice of pointers?

Comment (bsamek) [reply user_service_basic.go]: Fixed.

File: user_service_basic.go (hunk 2 above)
Comment (bsamek) [reply user_service_basic.go]: It doesn't have a password field. I could
1. Leave this as is.
2. Embed a basic user inside a naive user.
3. Add a password to the basic user.

Comment (tychoish) [reply user_service_basic.go]: I vote for 3

Comment (bsamek) [reply user_service_basic.go]: Done.

File: user_service_basic.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+// BasicUserManager implements the UserManager interface and has a list of
+// BasicUsers which is stored in the settings configuration file.
+// Note: This use of the UserManager is recommended for dev/test purposes only
+// and users who need high security authentication  mechanisms should rely on a
+// different authentication mechanism.
+type BasicUserManager struct {
+	users []basicUser
+}
+
+func NewBasicUserManager(users []basicUser) (UserManager, error) {
```
Comment (tychoish) [inline user_service_basic.go]: should just have a slice of users, and do type conversion here.

also docstring.

File: user_service_basic.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+)
+
+// BasicUserManager implements the UserManager interface and has a list of
+// BasicUsers which is passed into the constructor.
+type BasicUserManager struct {
+	users []basicUser
+}
+
+// NewBasicUserManager is a constructor to create a BasicUserManager from
+// a list of basic users. It requires the basicUser concrete type.
```
Comment (tychoish) [inline user_service_basic.go]: user created by the name of the constructor.

File: user_service_basic.go [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+func NewBasicUserManager(users []User) (UserManager, error) {
+	catcher := grip.NewBasicCatcher()
+	basicUsers := []basicUser{}
+	var bu *basicUser
+	var ok bool
+	for _, u := range users {
+		if bu, ok = u.(*basicUser); !ok {
+			catcher.Add(errors.Errorf("%T is not a basicUser", u))
```
Comment (tychoish) [inline user_service_basic.go]: catcher.Errorf

File: user_service_basic.go [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+			continue
+		}
+		basicUsers = append(basicUsers, *bu)
+	}
+	return &BasicUserManager{basicUsers}, nil
```
Comment (tychoish) [inline user_service_basic.go]: catcher.Resolve()

**Pull request merged**